website/docs/core-concepts/writing-code/workflows.md

```diff
@@ -2,155 +2,161 @@
+[Actions](/reference/appsmith-framework/widget-actions) in Appsmith are predefined functions that allow you to perform specific tasks in response to user interactions or other events in your app. 
 
-To read data by using an API: Create an API - `getAllUsers` on Appsmith by adding the `API` from `Explorer` >> Click `(+)` Query/JS >> Select `New Blank API` >> Rename to `getAllUsers` (or select an existing API) Add API call to the table data property of a table widget as shown in the code snippet below:
+They can be triggered by binding them to events, such as a button click. For example, you want to execute a API on button click and show a success message on the successful execution of the API. You can bind to invoke [showAlert()](/reference/appsmith-framework/widget-actions/show-alert) function on the `onClick()` event for `onSuccess` attribute, as shown below:
 
-```
- {{getAllUsers.data}}
+```javascript
+showAlert("Data saved successfully.");
 ```
 
-:::info
-The [table data property of a table](../../reference/widgets/table/#table-data) requires a **JSON Array** so verify that your API returns a **JSON Array**.
-:::
+<VideoEmbed host="youtube" videoId="tjJIDkoCyQE" title="Global Functions" caption="Global Functions"/> 
 
-A `{{API.run()}}` method, if supplied to the [Table data](../../reference/widgets/table/#table-data), will throw an error, as table data is a Sync field and cannot perform the execution. However, you can read the response generated by the API by accessing the data property `{{API.data}}` of an API. When you bound an API data property to a widget, Appsmith executes the API **on page load**. You can modify the API settings if you wish not to execute the API **on page load**.
+You can only use the **global actions** provided out-of-the-box by Appsmith in the **async** **fields**.  
 
-**Using a JSObject**
+##### Handle success or error
 
-You have a JSObject function that fetches all Users' information, filters data for `firstname` & `email,` and then returns the result so you can bind it to the table widget and display information.
+When configuring API and query executions, it's important to provide feedback to the user about the outcome of the action. This can be done by displaying success messages when an action is completed successfully or error messages when there are issues with the execution.
 
- <VideoEmbed host="youtube" videoId="8mVQS6uaR6M" title="Using a JSobject" caption="Using a JSobject"/>  
+<VideoEmbed host="youtube" videoId="4aEMFU1r1yg" title="Setup Success or Error messages" caption="Setup Success or Error messages"/>
 
+You can configure the action to be performed once a query returns with a success or an error. The success or error returned by the API or query can be determined by the HTTP status code or the query response status. These status code are bound to `onSuccess` and `onError` attributes of events and available in property pane to be setup for showing informative messages to end users.
```

Comment:
```suggestion
You can configure the action to be executed once a query returns with a success or an error. The success or error returned by the query can be determined by the HTTP status code or the query response status.
```

Reply:
@danciaclara - Why do we need to remove this - it gives developers an understanding how the callbacks are tied up to the properties available in property pane.

Reply:
@jnikhila Apologies. I think I may have misunderstood the context there. Can we modify it a little

> You can configure the action to be executed once a query returns with a success or an error. The success or error returned by the query can be determined by the HTTP status code or the query response status.  These status codes can be bound to the `onSuccess` and `onError` callback for showing informative messages to end users.

Reply:
The info we are providing is that when a query or API returns a HTTP status code say 200 - then that Status code is bound to `onSuccess` and this callback can be used to show the infromative message. Similarly, it's configured to handle HTTP error code generate by API/Query and in which case the `onError` can be used to show the message.